### PR TITLE
Mldb 1396 sdk api coverage

### PR DESCRIPTION
**Pull request by jeremybarnes**
This PR makes available more functionality from MLDB for plugins to use, and cleans up the APIs of that which is exposed; in particularly moving ML namespaced things to Datacratic, reducing include files and reducing the API surface to the minimum required for the functionality.

It was based upon observations of internal MLDB functionality required to build an external plugin.

There is also some dead code removal, which leads to a net reduction of 300 LOC.

There will be mechanical changes required to pro and other plugin repos.

### CODE REVIEW

**Comment by jeremybarnes** [comment]
(test failure is spurious; it's an external file from a service that is rate-limiting)

File: vfs/filter_streams.h [hunk 1]
```diff
@@ -299,8 +298,14 @@ class filter_istream : public std::istream {
     std::shared_ptr<void> sink;            ///< Ownership of streambuf
     std::atomic<bool> deferredFailure;
     std::string resource;
-    std::shared_ptr<Datacratic::FsObjectInfo> info_;
+    std::shared_ptr<FsObjectInfo> info_;
 };
 
+} // namespace Datacratic
 
+#if 0
```
Comment (guyd) [inline vfs/filter_streams.h]: Left over?

**Comment by guyd** [comment]
+1 - reviewed briefly

**Comment by jeremybarnes** [comment]
@mikeburkat Note that this PR requires changes to your plugin, but should allow you to link only against official APIs available in the MLDB SDK.

ML::filter_{i,o}stream -> Datacratic::filter_{i,o}stream
Possibly, some small changes to include more header files that were transitively included before

**Pull request merged**